lib/hammer_cli_foreman/commands.rb

```diff
@@ -157,7 +160,10 @@ def get_resource_ids(resource, options={})
       begin
         resolver.send("#{resource.singular_name}_ids", opts)
       rescue HammerCLIForeman::MissingSeachOptions => e
-        raise e unless (options[:required] == false)
+        if (options[:required] == true || resource_search_requested(resource, opts, plural=true))
```

Comment:
Please replace `plural=true` with 'true' only. It behaves a bit differently from what I think you expect.